internal/wrphandlers/qos/priority_queue.go

```diff
@@ -56,40 +61,44 @@ type item struct {
 	discard bool
 }
 
+func (itm *item) dispose() (payloadSize int64) {
+	payloadSize = int64(len(itm.msg.Payload))
+	// Mark itm to be discarded.
+	itm.discard = true
+	// Preemptively discard itm's payload to reduce
+	// resource usage, since itm will be discarded,
+	itm.msg.Payload = nil
+	itm.msg.RequestDeliveryResponse = &higherPriorityMessageTookTheSpot
+
+	return payloadSize
+}
+
 // Dequeue returns the next highest priority message.
-func (pq *priorityQueue) Dequeue() (wrp.Message, bool) {
-	var (
-		msg wrp.Message
-		ok  bool
-	)
-	for pq.Len() != 0 {
-		itm := heap.Pop(pq).(item)
-		// itm.discard will be true if `itm` has been marked to be discarded,
-		// i.e. trimmed by `pq.trim()'.
-		if itm.discard {
```

Comment:
Previously we were discarding the message without sending any response back to xmidt